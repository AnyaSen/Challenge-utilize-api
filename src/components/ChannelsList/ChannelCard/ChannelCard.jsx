import React from 'react';
import Styles from './ChannelCard.module.scss';

export default function ChannelCard({ image, description }) {
	return (
		<div data-testid='ChannelCard' className={Styles.ChannelCardContainer}>
			<img
				data-testid='ChannelImage'
				className={Styles.image}
				src={image}
				alt='Channel Logo'
			/>

			<p data-testid='ChannelDescription' className={Styles.description}>
				{description}
			</p>
		</div>
	);
}
