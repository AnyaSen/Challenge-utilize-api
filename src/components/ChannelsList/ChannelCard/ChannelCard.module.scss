@import '../../../assets/scss/breakpoints';
@import '../../../assets/scss/variables';

.ChannelCardContainer {
	display: flex;
	padding: $space-3 $space-4;
	border: 1px solid $color-border-color;
	border-radius: $border-radius;
	max-width: 45rem;
	margin: $space-2;
	min-width: 40%;

	@include sm {
		padding: $space-1 $space-2;
	}

	.image {
		height: 5rem;
		margin-right: $space-2;

		@include sm {
			height: 4rem;
		}
	}

	.description {
		margin-top: $space-4;
	}
}
