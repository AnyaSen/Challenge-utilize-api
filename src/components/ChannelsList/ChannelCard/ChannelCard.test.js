import React from 'react';
import { render, cleanup } from '@testing-library/react';

import ChannelCard from './ChannelCard';

describe('<ChannelCard />', () => {
	afterAll(cleanup);

	test('Should match props', () => {
		const props = {
			image: 'https://balbla.com',
			description: 'Some description..'
		};

		const { description, image } = props;

		const { getByText, getByAltText, container } = render(
			<ChannelCard {...props} />
		);

		const cardImage = getByAltText('Channel Logo');
		const descriptionText = getByText(description);

		expect(descriptionText).toBeTruthy();
		expect(cardImage.src).toContain(image);

		expect(container.firstChild).toMatchSnapshot();
	});
});
