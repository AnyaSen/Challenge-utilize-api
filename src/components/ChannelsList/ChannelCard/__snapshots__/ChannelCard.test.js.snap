// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChannelCard /> Should match props 1`] = `
<div
  class="ChannelCardContainer"
  data-testid="ChannelCard"
>
  <img
    alt="Channel Logo"
    class="image"
    data-testid="ChannelImage"
    src="https://balbla.com"
  />
  <p
    class="description"
    data-testid="ChannelDescription"
  >
    Some description..
  </p>
</div>
`;
